torch/testing/_internal/common_fsdp.py

```diff
@@ -60,10 +61,30 @@
     run_subtests,
     TEST_SKIPS,
 )
-from torch.testing._internal.common_utils import FILE_SCHEMA, get_cycles_per_ms
+from torch.testing._internal.common_utils import (
+    FILE_SCHEMA,
+    get_cycles_per_ms,
+    TEST_CUDA,
+    TEST_HPU,
+)
 from torch.utils._triton import has_triton
 
 
+DEVICE_COUNT = 4  # default
+
+if TEST_CUDA:
+    DEVICE_TYPE = "cuda"
+    DISTRIBUTED_BACKEND = "nccl"
+    DEVICE_COUNT = torch.cuda.device_count()
+elif TEST_HPU:
+    DEVICE_TYPE = "hpu:0"
+    DISTRIBUTED_BACKEND = "hccl"
```

Comment:
`DEVICE_TYPE = "hpu:0"` seems a bit hacky. Can you make it "hpu" and add a `DEVICE_COUNT` field?

Reply:
Another backend has the same "hccl" name.